src/config.rs

```diff
@@ -79,7 +79,12 @@ impl Context {
         let value = match key {
             Config::Selfavatar => {
                 let rel_path = self.sql.get_raw_config(self, key);
-                rel_path.map(|p| dc_get_abs_path(self, &p).to_str().unwrap().to_string())
+                rel_path.map(|p| {
+                    dc_get_abs_path(self, &p)
+                        .to_str()
+                        .unwrap_or_default()
+                        .to_string()
```

Comment:
I think you may want to replace this whole lot with `dc_get_abs_path(self, &p).to_string_lossy()` if you want to "safely" get on with things.